compiler/optimizer/src/main/java/org/apache/nemo/compiler/optimizer/pass/compiletime/annotating/ResourceSitePass.java

```diff
@@ -164,7 +164,7 @@ private static void assignNodeShares(
    */
   private static boolean isOneToOneEdge(final Collection<IREdge> inEdges) {
     return inEdges.size() == 1 && inEdges.iterator().next()
-      .getPropertyValue(CommunicationPatternProperty.class).get()
+      .getPropertyValue(CommunicationPatternProperty.class).orElseThrow(IllegalStateException::new)
```

Comment:
But in places like this, would it make less sense if this returns an exception if `ONE_TO_ONE` doesn't match an empty `Optional`, where the value is unassigned?

Reply:
In my thought, it makes sense only if the empty value of the `Optional` is legal. 

We suppose that all `IREdge` has `CommunicationPatternProperty` property value, so the empty value is an illegal state. In this case, it would be good to stop processing immediately and not propagate the illegal state.

Reply:
Alright, that makes sense. What do you think about the others, where the empty value of `Optional` is legal? Do you think we should make the changes there, or is the PR good to go?

Reply:
It seems that all `Optional`s changed in this PR must have some value but not empty. (I guess that this is the reason why the authors just `.get()` instead of `.orElseThrow(...)`.) If you found any of them may have a legal empty, please let me know.